components/higher-order/with-api-data/index.js

```diff
@@ -18,32 +18,26 @@ export default ( mapPropsToData ) => createHigherOrderComponent( ( WrappedCompon
 	class APIDataComponent extends Component {
 		constructor( props, context ) {
 			super( ...arguments );
-
-			this.state = {
-				dataProps: {},
-			};
-
 			this.schema = context.getAPISchema();
 			this.routeHelpers = mapValues( {
 				type: context.getAPIPostTypeRestBaseMapping(),
 				taxonomy: context.getAPITaxonomyRestBaseMapping(),
 			}, ( mapping ) => ( key ) => mapping[ key ] );
-		}
-
-		componentWillMount() {
+			this.state = {
+				dataProps: this.applyMapping( props ),
+			};
 			this.isStillMounted = true;
-			this.applyMapping( this.props );
 		}
 
 		componentDidMount() {
 			this.initializeFetchable( {} );
 		}
 
-		componentWillReceiveProps( nextProps ) {
-			this.applyMapping( nextProps );
-		}
-
 		componentDidUpdate( prevProps, prevState ) {
+			if ( prevProps !== this.props ) {
```

Comment:
Can we rely on this condition? I've no idea if React may be passing a new reference for props each time. Seems like `isShallowEqual` might be safer.

Reply:
I think we can rely on it (I tried), but It's a good suggestion since it can be a small perf improvement.